### PR TITLE
fix: placement subspecialty display logic

### PR DESCRIPTION
**Pull request by ReubenRobertsHEE**
Allow subspecialties to be shown if they exist, regardless of postAllowsSubspecialty.

TIS21-4983

### CODE REVIEW

**Review by DorisWongNHS — APPROVED** [review]

File: cypress/component/placements/Placements.cy.tsx [hunk 1]
```diff
@@ -248,7 +249,35 @@ describe("Placements with MFA set up", () => {
       .should("contain.text", "None provided");
   });
 
-  it("should not display Placements Subspecialty if Post does not Allow", () => {
+  it("should not display missing Placements Subspecialty if Post does not Allow", () => {
```
Comment (Judge40) [inline cypress/component/placements/Placements.cy.tsx:252]: These descriptions are the wrong way around I think?

**Review by Judge40 — CHANGES_REQUESTED** [review]

File: cypress/component/placements/Placements.cy.tsx (hunk 1 above)
Comment (ReubenRobertsHEE) [reply cypress/component/placements/Placements.cy.tsx:252]: Sorry, maybe poorly phrased. The intention was to test that placements without a subspecialty (i.e. 'missing') would not have the 'Sub specialty' field displayed at all if postAllowsSubspecialty = false

Comment (ReubenRobertsHEE) [reply cypress/component/placements/Placements.cy.tsx:252]: So not show 'Sub specialty: None provided' if postAllowsSubspecialty = false, basically.

Comment (Judge40) [reply cypress/component/placements/Placements.cy.tsx:252]: Ah, got it 🥴

Comment (ReubenRobertsHEE) [reply cypress/component/placements/Placements.cy.tsx:252]: I think I snuck in the merge just before your comment 😁  - if you like I can make a PR to fix the wording if you think its worth clarifying

Comment (ReubenRobertsHEE) [reply cypress/component/placements/Placements.cy.tsx:252]: Oh sorry, ok, looking at the code it is wrong - let me fix 🤦

Comment (Judge40) [reply cypress/component/placements/Placements.cy.tsx:252]: Actually, I'm not sure I do 😅 

e.g.
Description: "should display Placements Subspecialty even if Post does not Allow"
Mock data: `mockPlacementNoSubSpecialtyPostNotAllows`
Assertion: `cy.get('[data-cy="subSpecialty0Val"]').should("not.exist");`

The mock and the assertion match, but I can't get my head around the description of it "should display placements subspecialty" then asserting `not.exist`

**Pull request merged**